addons/mail/static/src/discuss/search/common/search_messages_panel.js

```diff
@@ -0,0 +1,74 @@
+/* @odoo-module */
+
+import { Component, useState } from "@odoo/owl";
+import { Message } from "@mail/core/common/message";
+import { SearchForm } from "@mail/core/common/search_form";
+import { useAutofocus, useService } from "@web/core/utils/hooks";
+import { _t } from "@web/core/l10n/translation";
+import { useStore } from "@mail/core/common/messaging_hook";
+
+export class SearchMessagesPanel extends Component {
+    static components = {
+        Message,
+        SearchForm,
+    };
+    static props = ["thread", "className?"];
+    static template = "mail.SearchMessagesPanel";
+
+    setup() {
+        this.store = useStore();
+        this.threadService = useService("mail.thread");
+        this.ui = useState(useService("ui"));
+        this.state = useState({
+            /** @type {import('@mail/core/message_model').Message[]}*/
+            searchMessages: [],
+            /** @type {string|undefined} */
+            searchTerm: undefined,
+        });
+        useAutofocus();
+    }
+
+    /**
+     * Highlight the given message and scrolls to it. In small mode, the
+     * pin menu is closed beforwards
+     *
+     * @param {import('@mail/core/message_model').Message} message
+     */
+    async onClickJump(message) {
+        if (this.ui.isSmall) {
+            // TODO: close the search menu
+            // this.env.pinMenu.close();
+            // Give the time to the pin menu to close before scrolling
+            // to the message.
+            await new Promise((resolve) => setTimeout(() => requestAnimationFrame(resolve)));
+        }
+        await this.env.messageHighlight?.highlightMessage(message.id, this.props.thread);
+    }
+
+    /**
+     * Get the message to display when nothing is pinned on this thread.
+     */
+    get emptyMessage() {
+        return _t("No messages found");
+    }
```

Comment:
This message is shown initially, when it should only be displayed when actively making a search and there are no messages.